core/modules/widgets/widget.js

```diff
@@ -232,10 +232,40 @@ Widget.prototype.hasVariable = function(name,value) {
 	return false;
 };
 
+/*
+Find the widget, walking up the widget-tree, that has a different transclusion variable than the current widget
+*/
+Widget.prototype.findParentTransclusionWidget = function() {
+	var node = this;
+	var transclusionVariable = this.getVariable("transclusion");
+	while(node.getVariable("transclusion") === transclusionVariable) {
+		node = node.parentWidget;
+	}
+	return node !== this ? node : null;
```

Comment:
please use braces here. Ternary operators are always used that way in the core. `(node !== this) ? node : null;` IMO it's more readable.